infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/TestSnapshotReader.java

```diff
@@ -49,13 +53,22 @@ public TestSnapshotReader(TestReaderConfiguration config) {
     @Override
     public SnapshotReadMessage read(UUID snapshotRequestId) {
         // Connect to endpoint
-        List<LogReplicationEntry> messages = new ArrayList<>();
+        List<LogReplicationEntryMsg> messages = new ArrayList<>();
 
         for (long i : seqNumsToRead) {
             Object data = runtime.getAddressSpaceView().read(i).getPayload(runtime);
-            LogReplicationEntryMetadata metadata = new LogReplicationEntryMetadata(MessageType.SNAPSHOT_MESSAGE,
-                topologyConfigId, i, baseSnapshot, snapshotRequestId);
-            messages.add(new LogReplicationEntry(metadata, (byte[]) data));
+            LogReplicationEntryMetadata metadata = LogReplicationEntryMetadata.newBuilder()
+                    .setEntryType(LogReplication.LogReplicationEntryType.SNAPSHOT_MESSAGE)
+                    .setSiteConfigID(topologyConfigId)
+                    .setTimestamp(i)
+                    .setSnapshotTimestamp(baseSnapshot)
+                    .setSyncRequestId(getUuidMsg(snapshotRequestId))
+                    .build();
+
```

Comment:
This goes along the lines of one of my other comments.

We could extract this function into `generateMetadataResponse(long, long, int, int, long...)` but the problem is that due to lack of type safety it is really easy to mess up the ordering, and assign a wrong value to a field. When using the builder directly, you know exactly what is being set since you are using the setter methods directly.